### ISSUE TITLE
Segmentation fault - using remote ssh to ubuntu from windows

### ISSUE THREAD

**Issue by jesusmogollon** [microsoft/vscode-remote-release#3515]
- VSCode Version: 1.48.0-insider
- Local OS Version: Windows 10 2004
- Remote OS Version: Ubuntu 18.04.05 x86_64
- Remote Extension/Connection Type: SSH/Docker/WSL: ssh to VirtualBox VM

Steps to Reproduce:
Installed Ubuntu 18.04 from mini.iso (minimum components)
Open visual studio code insider
Connect to host

vscode is unable to start the server with the following error:
```
[22:31:38.635] Log Level: 2
[22:31:38.644] remote-ssh@0.51.0
[22:31:38.644] win32 x64
[22:31:38.648] SSH Resolver called for "ssh-remote+ubuntu-mini", attempt 1
[22:31:38.648] SSH Resolver called for host: ubuntu-mini
[22:31:38.648] Setting up SSH remote "ubuntu-mini"
[22:31:38.682] Using commit id "db40434f562994116e5b21c24015a2e40b2504e6" and quality "insider" for server
[22:31:38.686] Install and start server if needed
[22:31:40.920] Checking ssh with "ssh -V"
[22:31:40.989] > OpenSSH_for_Windows_7.7p1, LibreSSL 2.6.5
[22:31:40.992] Running script with connection command: ssh -T -D 52923 ubuntu-mini bash
[22:31:40.999] Terminal shell path: C:\WINDOWS\System32\cmd.exe
[22:31:41.572] > jmogollon@ubuntu-mini's password:]0;C:\WINDOWS\System32\cmd.exe
[22:31:41.573] Got some output, clearing connection timeout
[22:31:41.575] Showing password prompt
[22:31:46.565] Got password response
[22:31:46.566] "install" wrote data to terminal: "******"
[22:31:46.582] > 
[22:31:47.392] > f26340257200: running
[22:31:47.457] > Acquiring lock on /home/jmogollon/.vscode-server-insiders/bin/db40434f562994116e
> 5b21c24015a2e40b2504e6/vscode-remote-lock.jmogollon.db40434f562994116e5b21c24015
> a2e40b2504e6
[22:31:47.466] > \ln /home/jmogollon/.vscode-server-insiders/bin/db40434f562994116e5b21c24015a2e4
> 0b2504e6/vscode-remote-lock.jmogollon.db40434f562994116e5b21c24015a2e40b2504e6.t
> arget /home/jmogollon/.vscode-server-insiders/bin/db40434f562994116e5b21c24015a2
> e40b2504e6/vscode-remote-lock.jmogollon.db40434f562994116e5b21c24015a2e40b2504e6
[22:31:47.484] > Found existing installation at /home/jmogollon/.vscode-server-insiders/bin/db404
> 34f562994116e5b21c24015a2e40b2504e6...
[22:31:47.511] > SSH_CONNECTION=192.168.10.128 52929 192.168.10.138 22      
> LANG=en_CA.UTF-8
> XDG_SESSION_ID=1
> USER=jmogollon
> PWD=/home/jmogollon
> HOME=/home/jmogollon
> SSH_CLIENT=192.168.10.128 52929 22
> MAIL=/var/mail/jmogollon
> SHELL=/bin/bash
> SHLVL=2
> VSCODE_AGENT_FOLDER=/home/jmogollon/.vscode-server-insiders
> LOGNAME=jmogollon
> XDG_RUNTIME_DIR=/run/user/1000
> PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/us
> r/local/games
> _=/usr/bin/printenv
> Starting server with command... /home/jmogollon/.vscode-server-insiders/bin/db40
[22:31:47.518] > 
> 434f562994116e5b21c24015a2e40b2504e6/server.sh --host=127.0.0.1 --enable-remote-
> auto-shutdown  --port=0 &> "/home/jmogollon/.vscode-server-insiders/.db40434f562
> 994116e5b21c24015a2e40b2504e6.log" < /dev/null
[22:31:47.530] > Waiting for server log...
[22:31:48.049] > Waiting for server log...
[22:31:48.577] > Waiting for server log...
[22:31:49.072] > Waiting for server log...
[22:31:49.585] > Waiting for server log...
[22:31:50.097] > Waiting for server log...
[22:31:50.611] > Waiting for server log...
[22:31:51.114] >  
> *
> * Reminder: You may only use this software with Visual Studio family products,  
> * as described in the license (https://go.microsoft.com/fwlink/?linkid=2077057) 
[22:31:51.121] > 
> *
> 
[22:31:51.134] > Server did not start successfully. Full server log at /home/jmogollon/.vscode-se
> rver-insiders/.db40434f562994116e5b21c24015a2e40b2504e6.log >>>
> Segmentation fault
[22:31:51.140] > 
> <<< End of server log
> f26340257200##32##
[22:31:51.141] Received install output: f26340257200##32##
[22:31:51.142] Resolver error: The VS Code Server failed to start
[22:31:51.149] ------




[22:31:52.170] "install" terminal command done
[22:31:52.171] Install terminal quit with output: 

1.
2.

<!-- Check to see if the problem is general, with a specific extension, or only happens when remote -->
Does this issue occur when you try this locally?: Yes/No
Does this issue occur when you try this locally and all extensions are disabled?: Yes/No
```

Trying to run the following command in the VM gives the following:

```
jmogollon@ubuntu-mini:~/.vscode-server-insiders/bin/db40434f562994116e5b21c24015a2e40b2504e6$ ./server.sh --host=127.0.0.1 --enable-remote-auto-shutdown  --port=0
Segmentation fault
```
Same error with the nightly build as well.
Not sure how to keep digging of what the problem, is any ideas?

**Comment by roblourens** [comment]
Does your mini.iso host meet these prereqs? https://code.visualstudio.com/docs/remote/linux#_remote-host-container-wsl-linux-prerequisites

**Comment by BrendanSimon** [comment]
I have the same problem with a Debian 10 Buster VirtualBox VM, which I have been using for months with VSCode Remote SSH.

The `node` binary in the `.vscode-server` directory segfaults immediately.  It's got to be something to do with the remote SSH extension surely.

Insiders has the same issue.

I had upgraded my Win 10 Pro OS to 2004 so I could play with WSL2 and also installed Docker.  I don't know if any of that is relevant, but thought I would mention it.

NOTE: I an successfully ssh to the Debian VM so it's not the SSH connection.

If I delete the `.vscode-server` directory on the remote, vscode will connect and reinstall it, but the same result.  segmentation fault.

**Comment by enjaxx** [comment]
I'm using the remote-ssh extension and had the same problem on a Buster vm. After removing the .vscode-server folder on the vm and enabling the telemetry gathering setting again I could download and execute the server on the vm again. I cloud reproduce this on current code and on insiders. WHY does telemetry gathering influence the installation of the vscode server on the vm???

**Comment by sickmon** [comment]
I am also facing same issue after recent visual studio update.

> cat .3dd905126b34dcd4de81fa624eb3a8cbe7485f13.log 
/home/vagrant/.vscode-server/bin/3dd905126b34dcd4de81fa624eb3a8cbe7485f13/server.sh: line 12:  4386 Segmentation fault      (core dumped) "$ROOT/node" ${INSPECT:-} "$ROOT/out/vs/server/main.js" "$@"


Tried removing /home/vagrant/.vscode-server and connecting again but it didn't helped. 


In VS GUI, 

[19:00:07.668] Log Level: 2
[19:00:07.671] remote-ssh@0.51.0
[19:00:07.671] win32 x64
[19:00:07.672] SSH Resolver called for "ssh-remote+127.0.0.1", attempt 1
[19:00:07.672] SSH Resolver called for host: 127.0.0.1
[19:00:07.673] Setting up SSH remote "127.0.0.1"
[19:00:07.694] Using commit id "3dd905126b34dcd4de81fa624eb3a8cbe7485f13" and quality "stable" for server
[19:00:07.695] Install and start server if needed
[19:00:07.720] Checking ssh with "ssh -V"
[19:00:07.809] > OpenSSH_for_Windows_7.7p1, LibreSSL 2.6.5
[19:00:07.812] Running script with connection command: ssh -T -D 64657 127.0.0.1 bash
[19:00:07.816] Terminal shell path: C:\Windows\System32\cmd.exe
[19:00:07.957] > 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> ]0;C:\Windows\System32\cmd.exe
[19:00:07.957] Got some output, clearing connection timeout
[19:00:07.987] > 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
[19:00:08.386] > 42d11b015e9a: running
> 
[19:00:08.502] > Acquiring lock on /home/vagrant/.vscode-server/bin/3dd905126b34dcd4de81fa624eb3a
> 8cbe7485f13/vscode-remote-lock.vagrant.3dd905126b34dcd4de81fa624eb3a8cbe7485f13 
> 
[19:00:08.512] > \ln /home/vagrant/.vscode-server/bin/3dd905126b34dcd4de81fa624eb3a8cbe7485f13/vs
> code-remote-lock.vagrant.3dd905126b34dcd4de81fa624eb3a8cbe7485f13.target /home/v
> agrant/.vscode-server/bin/3dd905126b34dcd4de81fa624eb3a8cbe7485f13/vscode-remote
> -lock.vagrant.3dd905126b34dcd4de81fa624eb3a8cbe7485f13
> 
[19:00:08.526] > Found existing installation at /home/vagrant/.vscode-server/bin/3dd905126b34dcd4
> de81fa624eb3a8cbe7485f13...
> 
[19:00:08.553] > LESSKEY=/etc/lesskey.bin
> MANPATH=
> NNTPSERVER=news
> XDG_SESSION_ID=8
> HOSTNAME=vagrant-sles-12-x64
> XKEYSYMDB=/usr/X11R6/lib/X11/XKeysymDB
> SHELL=/bin/bash
> HOST=vagrant-sles-12-x64
> HISTSIZE=1000
> PROFILEREAD=true
> SSH_CLIENT=10.0.2.2 64659 22
> MORE=-sl
> USER=vagrant
> XNLSPATH=/usr/X11R6/lib/X11/nls       
> HOSTTYPE=x86_64
> QEMU_AUDIO_DRV=pa
> FROM_HEADER=
> PAGER=less
> CSHEDIT=emacs
> VSCODE_AGENT_FOLDER=/home/vagrant/.vscode-server
[19:00:08.562] > 
> XDG_CONFIG_DIRS=/etc/xdg
> MINICOM=-c on
> LIBGL_DEBUG=quiet
> PATH=/usr/local/bin:/usr/bin:/bin:/usr/games:/usr/local/go/bin:/home/vagrant/gol
> ib/bin
> MAIL=/var/mail/vagrant
> CPU=x86_64
> PWD=/home/vagrant
[19:00:08.567] > 
> INPUTRC=/home/vagrant/.inputrc
> LANG=en_US.UTF-8
> PYTHONSTARTUP=/etc/pythonstart
> GPG_TTY=not a tty
> AUDIODRIVER=pulseaudio
> HOME=/home/vagrant
> SHLVL=2
> OSTYPE=linux
[19:00:08.574] > 
> LESS_ADVANCED_PREPROCESSOR=no
> SDL_AUDIODRIVER=pulse
> ALSA_CONFIG_PATH=/etc/alsa-pulse.conf
> GOROOT=/usr/local/go
> LOGNAME=vagrant
> MACHTYPE=x86_64-suse-linux
> LESS=-M -I -R
> G_FILENAME_ENCODING=@locale,UTF-8,ISO-8859-15,CP1252
[19:00:08.581] > 
> CVS_RSH=ssh
> SSH_CONNECTION=10.0.2.2 64659 10.0.2.15 22
> XDG_DATA_DIRS=/usr/share
> LESSOPEN=lessopen.sh %s
> GOPATH=/home/vagrant/golib:/home/vagrant/code
> XDG_RUNTIME_DIR=/run/user/1000
[19:00:08.587] > 
> LESSCLOSE=lessclose.sh %s %s
> G_BROKEN_FILENAMES=1
> COLORTERM=1
> _=/usr/bin/printenv
> Starting server with command... /home/vagrant/.vscode-server/bin/3dd905126b34dcd
> 4de81fa624eb3a8cbe7485f13/server.sh --host=127.0.0.1 --enable-remote-auto-shutdo
[19:00:08.593] > 
> wn  --port=0 &> "/home/vagrant/.vscode-server/.3dd905126b34dcd4de81fa624eb3a8cbe
> 7485f13.log" < /dev/null
> Waiting for server log...
[19:00:08.600] > 
> 
[19:00:09.104] > Waiting for server log...
> 
[19:00:09.619] > Waiting for server log...
> 
[19:00:10.135] > Waiting for server log...
> 
[19:00:10.645] > Waiting for server log...
> 
[19:00:11.176] > Waiting for server log...
> 
[19:00:11.697] > Waiting for server log...
> 
[19:00:12.233] >  
> *
> * Reminder: You may only use this software with Visual Studio family products,  
[19:00:12.238] > 
> * as described in the license (https://go.microsoft.com/fwlink/?linkid=2077057) 
> *
> 
> Server did not start successfully. Full server log at /home/vagrant/.vscode-serv
> er/.3dd905126b34dcd4de81fa624eb3a8cbe7485f13.log >>>
> /home/vagrant/.vscode-server/bin/3dd905126b34dcd4de81fa624eb3a8cbe7485f13/server
[19:00:12.244] > 
> .sh: line 12:  4679 Segmentation fault      (core dumped) "$ROOT/node" ${INSPECT
> :-} "$ROOT/out/vs/server/main.js" "$@"
> <<< End of server log
> 42d11b015e9a##32##
> 
[19:00:12.244] Received install output: 42d11b015e9a##32##
[19:00:12.245] Resolver error: The VS Code Server failed to start
[19:00:12.248] ------




[19:00:12.842] "install" terminal command done
[19:00:12.843] Install terminal quit with output:

**Comment by sickmon** [comment]
vagrant@vagrant-sles-12-x64:~/.vscode-server/bin/3dd905126b34dcd4de81fa624eb3a8cbe7485f13> ./server.sh 
internal/modules/cjs/loader.js:626
    throw err;
    ^

Error: Cannot find module '/home/vagrant/.vscode-server/bin/3dd905126b34dcd4de81fa624eb3a8cbe7485f13/out/vs/server/main.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:623:15)
    at Function.Module._load (internal/modules/cjs/loader.js:527:27)
    at Function.Module.runMain (internal/modules/cjs/loader.js:837:10)
    at internal/main/run_main_module.js:17:11 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
vagrant@vagrant-sles-12-x64:~/.vscode-server/bin/3dd905126b34dcd4de81fa624eb3a8cbe7485f13>

**Comment by roblourens** [comment]
It's likely that the remote does not meet our linux prereqs here: https://code.visualstudio.com/docs/remote/linux#_remote-host-container-wsl-linux-prerequisites

Or if the download was corrupted somehow you can try running the command "Kill VS Code Server on Host" and trying to connect again.

**Comment by BrendanSimon** [comment]
I checked my prequisites as listed int the above link.  All of them are installed (Debian 10 Buster).  Tried installing each pre-req with `apt` and it said they were all installed.

The only way I could get the vscode server to work was to manually download it onto the report and install it manually.

Removing everything and letting VSCode download and install doesn't work for me.

Not sure why this issue was closed based on an **assumption** of: **"likely"** something is not installed.
